src/translate.py

```diff
@@ -2236,38 +2245,31 @@ def add_imm(source: Expression, imm: Expression, stack_info: StackInfo) -> Expre
         # Pointer addition (this may miss some pointers that get detected later;
         # unfortunately that's hard to do anything about with mips_to_c's single-pass
         # architecture.
-        if isinstance(imm, Literal):
+        # If the imm is exactly a multiple of 0x10000, then it is likely that we are
+        # constructing a large offset (see deref)
+        if isinstance(imm, Literal) and imm.value % 0x10000 != 0:
```

Comment:
If a struct has a `char big[1000]` field that crosses the `0x10000`-byte boundary, the new `get_field()` is willing to return an access to the exact element at offset `0x10000`.

This PR ends up *fixing* some big struct accesses, like this one from OOT:
```diff
[INFO] Output of ../oot/asm/non_matchings/overlays/actors/ovl_Boss_Ganon2/func_808FD5F4.s changed! Diff:
--- 

+++ 

@@ -705,10 +705,10 @@

         }
         if (phi_v1_16 >= 0xE5U) {
             arg1->envCtx.unk_E1 = 1;
-            arg1->unk10B08 = 0xFF;
-            arg1->unk10B07 = 0xFF;
-            arg1->envCtx.unk_E2 = 0xFF;
-            arg1->unk10B09 = 0x64;
+            arg1->envCtx.unk_E2[2] = 0xFF;
+            arg1->envCtx.unk_E2[1] = 0xFF;
+            arg1->envCtx.unk_E2[0] = 0xFF;
+            arg1->envCtx.unk_E2[3] = 0x64;
             if (arg0->unk398 == 0xEA) {
                 arg1->envCtx.unk_E1 = 0;
                 arg0->unk39C = 0x18U;
```